src/core/service.c

```diff
@@ -3584,15 +3646,22 @@ static void service_sigchld_event(Unit *u, pid_t pid, int code, int status) {
 
 static int service_dispatch_timer(sd_event_source *source, usec_t usec, void *userdata) {
         Service *s = SERVICE(userdata);
+        const char *state_str = NULL;
 
         assert(s);
         assert(source == s->timer_event_source);
 
         switch (s->state) {
 
+        case SERVICE_CONDITION:
+                state_str = state_str ?: "Exec-condition";
+                _fallthrough_;
         case SERVICE_START_PRE:
+                state_str = state_str ?: "Start-pre";
+                _fallthrough_;
         case SERVICE_START:
-                log_unit_warning(UNIT(s), "%s operation timed out. Terminating.", s->state == SERVICE_START ? "Start" : "Start-pre");
+                state_str = state_str ?: "Start";
+                log_unit_warning(UNIT(s), "%s operation timed out. Terminating.", state_str);
```

Comment:
doesn't matter much, but maybe just do this instead:

```c
log_unit_warning(UNIT(s), "Operation '%s' timed out. Terminating.", service_state_to_string(s->state));
```
I figure this is similarly descriptive, but not that ugly in code...